setup.py

```diff
@@ -45,4 +45,5 @@ def calculate_version():
         'Topic :: Scientific/Engineering :: Artificial Intelligence'
     ],
     keywords=['pipeline optimization', 'hyperparameter optimization', 'data science', 'machine learning', 'genetic programming', 'evolutionary computation'],
+    extra_require = {'xgboost': ['xgboost']},
```

Comment:
I believe it's `extras_require`: http://peak.telecommunity.com/DevCenter/setuptools#declaring-extras-optional-features-with-their-own-dependencies